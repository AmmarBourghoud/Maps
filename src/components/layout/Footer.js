import React from "react";
import { AppBar, Container, Toolbar, Typography, Link } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
    footer: {
      bottom: '0',
      marginTop: '60%',
    }, 
  }));

export default function Footer() {
    const classes = useStyles();
    return (
        <footer>
        <AppBar position="sticky" className={classes.footer} color="transparent">
          <Container maxWidth="md">
            <Toolbar>
                <Typography variant="caption" color="textSecondary" align="left" >
                <Link color="primary" href="https://github.com/AmmarBourghoud/namR" target="_blank">
                     abourgho
                </Link>{' '}
                     {new Date().getFullYear()}
                       {'.'}
                </Typography>
            </Toolbar>
          </Container>
        </AppBar>
        </footer>
    )
}
