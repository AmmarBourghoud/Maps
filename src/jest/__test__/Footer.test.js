import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import Footer from '../../components/layout/Footer';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders Login component without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Footer/>, div); 
  ReactDOM.unmountComponentAtNode(div);
});

it('renders button correctly', () => {
  const {getByTestId} = render(<Footer />);
  expect(getByTestId('link')).toHaveTextContent("abourgho"); 
});

it('matches snapshot', () => {
  const tree = renderer.create(<Footer />).toJSON(); 
  expect(tree).toMatchSnapshot(); 
}); 

it("should go to the correct link", () => {
  const utils = render(<Footer />)
  const link = utils.getByTestId('link')
  expect(link.closest('a')).toHaveAttribute('href', "https://github.com/AmmarBourghoud/namR")
});