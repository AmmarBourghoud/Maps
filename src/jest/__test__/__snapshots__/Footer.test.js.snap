// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<footer>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorTransparent makeStyles-footer-3 MuiPaper-elevation4"
  >
    <div
      className="MuiContainer-root MuiContainer-maxWidthMd"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <span
          className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-alignLeft"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            data-testid="link"
            href="https://github.com/AmmarBourghoud/namR"
            onBlur={[Function]}
            onFocus={[Function]}
            target="_blank"
          >
            abourgho
          </a>
           
          2021
          .
        </span>
      </div>
    </div>
  </header>
</footer>
`;
